### PR TITLE
feat: fire session archive events for statistics when session gets ar…

### PR DESCRIPTION
**Pull request by tkuzynow**
…chived

Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: src/main/java/de/caritas/cob/userservice/api/service/statistics/event/ArchiveStatisticsEvent.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import de.caritas.cob.userservice.statisticsservice.generated.web.model.EventType;
+import de.caritas.cob.userservice.statisticsservice.generated.web.model.UserRole;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ArchiveStatisticsEvent implements StatisticsEvent {
```
Comment (patric-dosch-vi) [inline src/main/java/de/caritas/cob/userservice/api/service/statistics/event/ArchiveStatisticsEvent.java:21]: I would add a test like the others... see AssignSessionStatisticsEventTest

File: src/test/java/de/caritas/cob/userservice/api/service/archive/SessionArchiveServiceTest.java [hunk 2]
```diff
@@ -113,11 +121,18 @@ public void archiveSession_Should_ChangeStatusOfSession_WhenUserHasPermission()
 
     Session session = Mockito.mock(Session.class);
     when(session.isAdvised(any())).thenReturn(true);
+    when(session.getUser()).thenReturn(new User());
     when(sessionRepository.findById(SESSION_ID)).thenReturn(Optional.of(session));
 
     sessionArchiveService.archiveSession(SESSION_ID);
 
     verify(session, times(1)).setStatus(SessionStatus.IN_ARCHIVE);
+    ArgumentCaptor<ArchiveStatisticsEvent> statisticsEventArgumentCaptor =
```
Comment (patric-dosch-vi) [inline src/test/java/de/caritas/cob/userservice/api/service/archive/SessionArchiveServiceTest.java:141]: I personally would add a 2nd test that asserts the handling of the event stuff.

**Comment by patric-dosch-vi** [comment]
hey @tkuzynow, I took a quick look and a test is failing :/

File: src/main/java/de/caritas/cob/userservice/api/service/statistics/event/ArchiveStatisticsEvent.java (hunk 1 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/userservice/api/service/statistics/event/ArchiveStatisticsEvent.java:21]: added

File: src/test/java/de/caritas/cob/userservice/api/service/archive/SessionArchiveServiceTest.java (hunk 2 above)
Comment (tkuzynow) [reply src/test/java/de/caritas/cob/userservice/api/service/archive/SessionArchiveServiceTest.java:141]: added second test for exception scenario

Comment (patric-dosch-vi) [reply src/test/java/de/caritas/cob/userservice/api/service/archive/SessionArchiveServiceTest.java:141]: I should write a bit more carefully my review comments.
I had meant that to the currently existing test "archiveSession_Should_ChangeStatusOfSession_WhenUserHasPermission" I would write a second test that takes care of the asserts regarding the "fireArchiveEvent(session);". After all there are two different things happening here, once we test archiving and once we test sending the event. I wouldn't mix that in one test. And if I did, at least the test name would have to be adjusted.

ps. I did not thought about the "Catch" part.

**Review by patric-dosch-vi — APPROVED** [review]

File: src/test/java/de/caritas/cob/userservice/api/service/archive/SessionArchiveServiceTest.java (hunk 2 above)
Comment (tkuzynow) [reply src/test/java/de/caritas/cob/userservice/api/service/archive/SessionArchiveServiceTest.java:141]: hm, we are testing public method
, and we want to make sure that when we "archive" a session both things happen, session status get's changed AND event should be fired. I don't see added value of having it in separate test methods TBH. I agree to renaming the test to reflect it.

Comment (patric-dosch-vi) [reply src/test/java/de/caritas/cob/userservice/api/service/archive/SessionArchiveServiceTest.java:141]: then we go with the name

**Pull request merged**